tests/test_lifespan_fixture_class.py

```diff
@@ -0,0 +1,75 @@
+import asyncio
+from collections.abc import AsyncGenerator, Generator
+
+import pytest
+import pytest_asyncio
+
+from pytest_fixture_classes import fixture_class
+
+execution_log: list[str] = []
```

Comment:
Does it maybe make sense to keep the execution log per test? The whole clearing thing feels like an anti-pattern here, removing the ability to parallelize tests. It isn't a big problem for this library but I do feel like a testing library should be promoting the right patterns.